### ISSUE TITLE
Small object problem

### ISSUE THREAD

**Issue by tkuanlun350** [TuSimple/mx-maskrcnn#60]
Hi ! 

Currently my training process sometimes get RCNNL1-loss = nan when training maskrcnn.
I used your gist program to inspect my mask and image and seemed no problem!

My dataset only contain small object, therefore when testing rpn the size after 50-100 are all nan, is that the expected results ? (All bbox in my data are small )

The original resolution of my image is [200*200], do you think that's a problem ? or should I resize to the scale similar to Cityscape ? Can you give me some suggestion ? Thank you very much.

**Comment by Zehaos** [comment]
Hi, @tkuanlun350 
It is normal of your rpn result.
If you want to detect small object only, there is no need to use the original version of FPN. You can use P2 only (we use [P2, P3, P4, P5, P6] in RPN and [P2, P3, P4, P5] in RCNN ). 
The original version of FPN have requirement for the size of the input image (think of downsample times).

**Comment by tkuanlun350** [comment]
@Zehaos  Thanks for your fast reply ! I am working on scaling my raw image and the training process seemed to be stable now :)

Using P2 is a great advise ! I will definitely run some experiments.
